src/recaplog

```diff
@@ -20,145 +20,252 @@
 EMAIL_ON_ERROR=1
+EMAIL_ON_ERROR_SUBJECT="recaplog error"
 EMAIL_ON_ERROR_RECIPIENTS="root@localhost"
 
-###############################
-########## FUNCTIONS ##########
-###############################
+## FUNCTIONS ##
+
+# Function to generate timestamps
+ts() {
+  TS_FLAGS='--rfc-3339=seconds'
+  date ${TS_FLAGS}
+}
 
-# print message facility with pre-defined error-levels
-print_msg() {
-	if [ "$1" == "DEBUG" ]; then
-		PREFIX="[DD]"
-	elif [ "$1" == "INFORMATION" ]; then
-                PREFIX="[II]"
-	elif [ "$1" == "WARNING" ]; then
-                PREFIX="[WW]"
-	elif [ "$1" == "ERROR" ]; then
-                PREFIX="[EE]"
-	elif [ "$1" == "CRITICAL" ]; then
-                PREFIX="[CC]"
-	fi
-	if [[ $1 -le $ERRORLEVEL ]]; then
-		echo ${PREFIX}" "$2 | tee -a $LOGFILE
-	fi
-	# send email?
-	if [ $1 == "ERROR" -o $1 == "CRITICAL" ]; then
-		if [ $EMAIL_ON_ERROR -eq 1 ]; then
-			echo ${PREFIX}" "$2 | mutt -s $EMAIL_ON_ERROR_SUBJECT $EMAIL_ON_ERROR_RECIPIENTS
-		fi
-	fi
+# Function to log messages
+log() {
+  # does not work in a while-loop as spawns a new shell
+  local msg_type=$1
+  shift
+  local log_entry="$*"
+  ## This avoids sending any output to stdout when executed through cron
+  ## is helpful to avoid emails submitted, instead the logs contain the possible ERRORS
+  if ! tty -s; then
+    echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 >> ${LOGFILE}
+  else
+    if [[ ${VERBOSE} ]]; then
+      echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 | tee -a ${LOGFILE}
+    else
+      if [[ ${msg_type} =~ "ERROR" ||
+            ${msg_type} =~ "WARNING" ]]; then
+        echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 | tee -a ${LOGFILE}
+      else
+        echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 >> ${LOGFILE}
+      fi
+    fi
+  fi
 }
 
-# delete log files older than LOG_EXPIRY days
+# Function to delete log files older than LOG_EXPIRY days
 delete_old_logs() {
-	if [ $LOG_EXPIRY -eq 0 ]; then
-		print_msg INFORMATION "skipping old log file expiration..."
-	else
-		print_msg INFORMATION "deleting log files older than ${LOG_EXPIRY} days..."
-		# delete *.log files
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log" -type f -mtime +$LOG_EXPIRY -exec echo rm {} \;  | tee -a $LOGFILE
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log" -type f -mtime +$LOG_EXPIRY -exec rm {} \;
-		# delete *.log.gz files
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log.gz" -type f -mtime +$LOG_EXPIRY -exec echo rm {} \;  | tee -a $LOGFILE
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log.gz" -type f -mtime +$LOG_EXPIRY -exec rm {} \;
-	fi
+  local prefix_msg='Deleting:'
+  local matched_files=''
+  local empty_dirs=''
+  local count_files=0
+  local count_dirs=0
+  if [[ ${LOG_EXPIRY} -eq 0 ]]; then
+    log INFO "Skipping old log file expiration..."
```

Comment:
Add a "return" here and close your 'if' block.  In general, if your conditional block goes to the end of the function, just put a conditional 'return' at the start instead.  It minimizes indentation and makes the code easier to read.